### ISSUE TITLE
Access to first index of an Split and later make another Split, throws the error "Object is possibly 'undefined'"

### ISSUE THREAD

**Issue by RodrigoTomeES** [microsoft/TypeScript#48374]
# Bug Report

Hi,

I am triying to get the extension file inside of an URL like: 

 - https://example.com/folder/file.jpg: jpg
 - https://example.com/fold.er/fil.e.jpg?param.eter#hash=12.345: jpg

### 🔎 Search Terms

Object is possibly 'undefined'
undefined

### 🕗 Version & Regression Information

- I tried it in 4.5.4 version

### ⏯ Playground Link

[Playground link with relevant code](https://www.typescriptlang.org/play?#code/KYDwDg9gTgLgBAYwgOwM7wObBgVSgGwFEQZg0BLFOAXjgAoBXAgLjnSnOQwEpX3OMNAHwAoOHCb4AdKjD5yMOgHoA2gGIA-AF0l3FQAYtMuQroByKWe5TIYOtZgcAtvYDcQA)

### 💻 Code

```ts
export const getUrlExtension = (url: string): string =>
  url.split(/[#?]/)[0].split('.').pop().trim();
```

### 🙁 Actual behavior

Throw this error: `Object is possibly 'undefined'.`

### 🙂 Expected behavior

Doesn't throw any error, I think that never can be undefined.

**Comment by RodrigoTomeES** [comment]
@MartinJohns thanks! and sorry to duplicate the issue but I didn't find it
